Godeps/_workspace/src/k8s.io/kubernetes/pkg/storage/etcd/etcd_helper.go

```diff
@@ -264,6 +264,20 @@ func (h *etcdHelper) Delete(ctx context.Context, key string, out runtime.Object)
 }
 
 // Implements storage.Interface.
+func (h *etcdHelper) DeleteDir(ctx context.Context, directory string) error {
+	if ctx == nil {
+		glog.Errorf("Context is nil")
```

Comment:
@liggitt  Wouldn't this make the behaviour of DeleteDir different from the rest of the methods, e.g. Watch?